main.cpp

```diff
@@ -129,7 +129,13 @@ void ProcessMidiEvent(u32 dwParam1, bool mouse) {
         int chanbit=1<<midichan;
         msg.msg=(unsigned char)dwParam1;
         msg.data1=(unsigned char)(dwParam1>>8);
-        msg.data2=(unsigned char)(dwParam1>>16);		
+        msg.data2=(unsigned char)(dwParam1>>16);
+
+		// remap zero-velocity note-on events to note-off ones
+		if (msg.msg==0x90 && msg.data2 == 0) {
+			msg.msg=0x80;
```

Comment:
actually this is *nearly* right. I think you have to mask out the lo nybble (channel) to make this work for channels other than channel 0. ie if miditype==9 && msg.data2==0, msg.msg-=0x10; or something similar would do it.